### PR TITLE
extractor: added retry to mitigate connection errors

### PR DESCRIPTION
**Pull request by jstucke**
`ConnectionError`s mostly appear in tests because the docker container is not yet ready for accepting extraction requests

### CODE REVIEW

File: src/unpacker/extraction_container.py [hunk 1]
```diff
@@ -4,17 +4,21 @@
...
 from docker.errors import APIError, DockerException
 from docker.models.containers import Container
 from docker.types import Mount
+from requests.adapters import HTTPAdapter, Retry
 
 from config import cfg
 
 DOCKER_CLIENT = docker.from_env()
 EXTRACTOR_DOCKER_IMAGE = 'fkiecad/fact_extractor'
+retries = Retry(total=5, backoff_factor=0.1)
```
Comment (maringuu) [inline src/unpacker/extraction_container.py:21]: Why not put this directly in the argument? It is only used once.

File: src/unpacker/extraction_container.py [hunk 2]
```diff
@@ -24,6 +28,8 @@ def __init__(self, id_: int, tmp_dir: TemporaryDirectory, value: multiprocessing
         self.port = cfg.unpack.base_port + id_
         self.container_id = None
         self.exception = value
+        self.session = requests.Session()
```
Comment (maringuu) [inline src/unpacker/extraction_container.py:31]: This should be a private variable.

File: src/unpacker/extraction_container.py [hunk 3]
```diff
@@ -24,6 +28,8 @@ def __init__(self, id_: int, tmp_dir: TemporaryDirectory, value: multiprocessing
...
+        self.session.mount('http://', HTTPAdapter(max_retries=retries))
```
Comment (maringuu) [inline src/unpacker/extraction_container.py:32]: Shouldn't sessions be closed?

**Review by maringuu — CHANGES_REQUESTED** [review]

File: src/unpacker/extraction_container.py (hunk 1 above)
Comment (jstucke) [reply src/unpacker/extraction_container.py]: in-lined it

File: src/unpacker/extraction_container.py (hunk 2 above)
Comment (jstucke) [reply src/unpacker/extraction_container.py]: switched to usage as context manager

File: src/unpacker/extraction_container.py (hunk 3 above)
Comment (jstucke) [reply src/unpacker/extraction_container.py]: switched to usage as context manager

**Review by maringuu — APPROVED** [review]
LGTM!

**Pull request merged**